src/html-transform.ts

```diff
@@ -0,0 +1,256 @@
+    if (src.includes('web-component-tester/browser.js') ||
+        src.includes('wct-browser-legacy/browser.js')) {
+      wctScript = script;
+    }
+  }
+
+  // Script tags that aren't getting the AMD module treatment.
+  const standardTransformScripts = [...otherScripts];
+  let requireJsScript;
+
+  if (shouldTransformEsModuleToAmd) {
+    transformEsModulesToAmd(moduleScripts, options.js);
+
+    if (options.isEntryPoint && moduleScripts.length > 0) {
+      // We need RequireJS to load the AMD modules we are declaring. Inject the
+      // dependency as late as possible (just before the first module is
+      // declared) because there may be some UMD dependencies that we want to
+      // continue to load in global mode instead of AMD mode (which is detected
+      // by the presence of the `require` global).
+      const fragment = parse5.parseFragment(
+          `<script src="${options.requireJsUrl}"></script>\n`);
+      requireJsScript = fragment.childNodes![0];
+      const firstModule = moduleScripts[0];
+      dom5.insertBefore(firstModule.parentNode!, firstModule, fragment);
+    }
+
+  } else {
+    standardTransformScripts.push(...moduleScripts);
+  }
+
+  for (const script of standardTransformScripts) {
+    if (dom5.hasAttribute(script, 'src')) {
+      continue;
+    }
+    dom5.setTextContent(
+        script,
+        jsTransform(
+            dom5.getTextContent(script),
+            {...options.js, transformEsModulesToAmd: false}));
+  }
+
+  if (wctScript && requireJsScript) {
+    addWctTimingHack(wctScript, requireJsScript);
+  }
+
+  return parse5.serialize(document);
+}
+
+function transformEsModulesToAmd(
+    moduleScripts: dom5.Node[], jsOptions: JsTransformOptions|undefined) {
+  let previousGeneratedModule: string|undefined;
+
+  for (let i = 0; i < moduleScripts.length; i++) {
+    const script = moduleScripts[i];
+
+    // Module scripts execute in order. AMD modules don't necessarily preserve
+    // this ordering. To emulate the ordering, we construct an artificial
+    // dependency chain between all module scripts on the page.
+    const generatedModule = `polymer-build-generated-module-${i}`;
+
+    const isExternal = dom5.hasAttribute(script, 'src');
+    if (isExternal) {
+      const deps = [];
+      if (previousGeneratedModule !== undefined) {
+        deps.push(previousGeneratedModule);
+      }
+      const externalSrc = dom5.getAttribute(script, 'src');
+      deps.push(externalSrc);
+      const depsStr = deps.map((dep) => `'${dep}'`).join(', ');
+      dom5.removeAttribute(script, 'src');
+      dom5.setTextContent(
+          script, `define('${generatedModule}', [${depsStr}]);`);
+
+    } else {
+      // Transform inline scripts with the AMD Babel plugin transformer.
+      const depStr = previousGeneratedModule === undefined ?
+          '' :
+          `'${previousGeneratedModule}', `;
+      dom5.setTextContent(
+          script,
+          jsTransform(
```

Comment:
style nit: I think this might read more linearly and format nicer if you put the new source in a variable

Reply:
Done.